state/upgrades.go

```diff
@@ -1405,30 +1405,42 @@ func DeleteCloudImageMetadata(st *State) error {
 // CopyMongoSpaceToHASpaceConfig copies the Mongo space name from
 // ControllerInfo to the HA space name in ControllerConfig.
 // This only happens if the Mongo space state is valid, it is not empty,
-// and if the is no value already set for the HA space name.
-func CopyMongoSpaceToHASpaceConfig(st *State) error {
-	info, err := st.ControllerInfo()
-	if err != nil {
-		return errors.Annotate(err, "cannot get controller info")
-	}
-
-	ms := info.MongoSpaceName
-	if info.MongoSpaceState != MongoSpaceValid || ms == "" {
-		return nil
+// and if there is no value already set for the HA space name.
+// The old keys are then deleted from ControllerInfo.
+func MoveMongoSpaceToHASpaceConfig(st *State) error {
+	// Holds Mongo space fields removed from controllersDoc.
+	type controllersUpgradeDoc struct {
+		MongoSpaceName  string `bson:"mongo-space-name"`
+		MongoSpaceState string `bson:"mongo-space-state"`
 	}
+	var doc controllersUpgradeDoc
```

Comment:
I'm fine with creating a type, but I think:

var doc struct {
  MongoSpaceName string `bson:...`
}

works as well.
That said, errors/debugging info is usually nicer if you have a type name, I think.